src/components/price-color-legend.tsx

```diff
@@ -33,7 +33,9 @@ export const MapPriceColorLegend = ({
   id: string;
 }) => {
   const formatCurrency = useFormatCurrency();
-  const [open, setOpen] = useState(true);
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down("md"));
```

Comment:
Can you extract this into a useIsMobile helper ? This way we cannot make a mistake of which breakpoint to use in other files.